lib/rpc_lwt.ml

```diff
@@ -97,19 +107,27 @@ module GenServer () = struct
       | _ -> Lwt.fail (MarshalError "All arguments must be named currently")
     in
 
-    let get_arg args name =
-      if not (List.mem_assoc name args)
-      then Lwt.fail (MarshalError (Printf.sprintf "Argument missing: %s" name))
-      else Lwt.return (List.assoc name args)
+    let get_arg args name is_opt =
+      if is_opt
+      then begin
+        if List.mem_assoc name args
+        then Rpc.Enum [List.assoc name args]
+        else Rpc.Enum []
+      end else begin
+        if List.mem_assoc name args
+        then List.assoc name args
+        else raise (MarshalError (Printf.sprintf "Argument missing: %s" name))
+      end
```

Comment:
`Lwt.return` and `Lwt.fails` is missing in a few places in this function

Reply:
ah oops, cut-n-pasted this from here to Async, fixed return types there (non monadic), fixed a bug, cut-n-pasted it back, boom!

We need to reduce code duplication in this library :-(